### ISSUE TITLE
Caching problem when used twice on a page

### ISSUE THREAD

**Issue by smichaelsen** [helhum/typoscript_rendering#16]
In two different plugins I use the `t:uri.ajaxAction` ViewHelper and it works great in both of them. But as soon as the page is cached, the second usage breaks, because then the viewhelper always returns the result from the first call again. For example:

Uncached page rendering:
```
<div class="my-first-plugin" data-ajaxurl="?this=is&the=url&for=the&first=plugin"></div>
<!-- ... -->
<div class="my-second-plugin" data-ajaxurl="?this=is&the=url&for=the&second=plugin"></div>
```

Cached page rendering (notice the identical urls):
```
<div class="my-first-plugin" data-ajaxurl="?this=is&the=url&for=the&first=plugin"></div>
<!-- ... -->
<div class="my-second-plugin" data-ajaxurl="?this=is&the=url&for=the&first=plugin"></div>
```

I didn't really manage to track the problem down - I'm not sure wether typoscript_rendering is the problem here. But if you have any ideas, I'd appreciate it very much.

**Comment by helhum** [comment]
Without any further hints how to reproduce this issue, I cannot really help.

Please also provide more information like: what TYPO3 version you exactly use and how exactly your plugin setup is (what plugins (extbase or not), does the URI have a cHash, how is the cached/ uncached page exactly requested (which uri/ cHash or not) etc.

**Comment by helhum** [comment]
Tried to reproduce this in current TYPO3 master with 2 different plugins from different extensions. Links were generated just fine. How exactly does your `t:uri.ajaxAction` usage looks like? If possible, please provide actual code, not pseudo code.

**Comment by smichaelsen** [comment]
Sorry for bothering. My problem is still present, but from digging down I can rule out your extension as the cause.

**Comment by smichaelsen** [comment]
For the record, here are my findings.
I used the ViewHelper like this:
`{t:uri.ajaxAction(action:'ajaxSingle',controller:'BookingForm',pluginName:'AdventureSingle',format:'json',arguments:'{editBooking:editBooking}') -> f:format.htmlentities()}`

And suprisingly the HtmlentitiesViewHelper was involved in the problem. When it was initialized by Fluid for the second time it had the same `$renderChildrenClosure` like on the first run, so it rendered the first AjaxActionViewHelper again. No idea why exactly that happens. I got rid of the  `-> f:format.htmlentities()` and now the problem is gone.
